### PR TITLE
Fix compilation with calloop 0.4

### PR DESCRIPTION
**Pull request by pablostebler**
I also needed to add a few Debug `impl`s in calloop. https://github.com/Smithay/calloop/pull/5

### CODE REVIEW

**Comment by vberger** [comment]
Hi, thanks for the pull request!

However, `InsertError` implements `Into<IoError>`, and I'd rather do the conversion rather than change the prototype for these functions.

There is work underway to significantly refactor this code, so I'd like to change it as little as possible in the meantime.

File: Cargo.toml [hunk 1]
```diff
@@ -18,7 +18,7 @@ tempfile = "2.1.5"
 slog = "2.1.1"
 slog-stdlog = "3.0.2"
 libloading = "0.4.0"
-wayland-client = { version = "0.20.5", optional = true }
+wayland-client = { version = "0.21.1", features = ["egl"], optional = true }
```
Comment (vberger) [inline Cargo.toml]: I don't understand, why did you need to add this feature here?

Comment (pablostebler) [reply Cargo.toml]: It's no longer a default feature since 0.21. But in fact I suppose that there is a good reason not to use 0.21 so I should probably remove this change from my PR.

Comment (vberger) [reply Cargo.toml]: Ah right, I forgot about that. Okay! :+1: 

Upgrading wayland-client is completely fine, but in this case, could you also upgrade winit to 0.18 ? I believe the upgrade should be trivial.

Comment (pablostebler) [reply Cargo.toml]: Yes, it just compiles and anvil seems to work normally.

**Comment by vberger** [comment]
From the travis build, it looks like there is some code in the logind backend that needs updating, too.

**Comment by vberger** [comment]
Hi, thanks this looks good! :+1: 

There is just a small fmt issue before I can merge, see the travis build.

**Comment by vberger** [comment]
Perfect, thanks!
